src/mame/video/hng64_sprite.hxx

```diff
@@ -46,6 +46,8 @@ void hng64_state::draw_sprites(screen_device &screen, bitmap_rgb32 &bitmap, cons
 	uint32_t *source = m_spriteram;
 	uint32_t *finish = m_spriteram + 0xc000/4;
 
+	std::multimap<int, uint32_t*> sortlist;
```

Comment:
Is a multimap the best structure for this?  It’s going to cause one allocation per element.  You might be better off using a vector, pre-allocating, and using either `lower_bound` or `upper_bound` to find the right place to insert things.

Reply:
The list does have a predetermined maximum size, but needs to allow multiple elements to have the same key.

Reply:
Yeah, but with a vector you only need to do one allocation, you can choose whether you want to insert elements with the same key at the first or last valid location for that key (by using `lower_bound` or `upper_bound` to choose the location), you still have the same O(log2(n)) performance to find the insert location, you have better locality, and the O(n) “slide” when inserting at a location other than the end is pretty cheap when the elements are trivially copyable.